shell/browser/ui/devtools_ui.cc

```diff
@@ -68,13 +99,30 @@ class BundledDataSource : public content::URLDataSource {
     // Serve request from local bundle.
     std::string bundled_path_prefix(kChromeUIDevToolsBundledPath);
     bundled_path_prefix += "/";
+    LOG(ERROR) << "Got Request: " << url.spec();
     if (base::StartsWith(path, bundled_path_prefix,
                          base::CompareCase::INSENSITIVE_ASCII)) {
       StartBundledDataRequest(path.substr(bundled_path_prefix.length()),
                               std::move(callback));
       return;
     }
 
+    std::string remote_path_prefix(kChromeUIDevToolsRemotePath);
+    remote_path_prefix += "/";
+    if (base::StartsWith(path, remote_path_prefix,
+                         base::CompareCase::INSENSITIVE_ASCII)) {
+      GURL url(kRemoteFrontendBase + path.substr(remote_path_prefix.length()));
+
+      CHECK_EQ(url.host(), kRemoteFrontendDomain);
+      if (url.is_valid() && DevToolsUIBindings::IsValidRemoteFrontendURL(url)) {
```

Comment:
Our `inspectable_web_contents_impl` is based on `devtools_ui_bindings` , the sanitization logic can be copied there. We already have a couple of copied code path for devtools ui actions, so wouldn't hurt to have this one part of it.

Reply:
@deepak1556 It's a decent chunk of code 🤷‍♂ If we're ok with copying it over I can do that 👍